Attempting to make under Arch Linux fails

Attempting to compile the project from source (or from the AUR, which I initially tried before going straight to source) gives the following, resulting in the program failing to compile:
```
[  0%] Built target zdl_autogen_timestamp_deps
[  3%] Automatic MOC for target zdl
[  3%] Built target zdl_autogen
[  6%] Building CXX object CMakeFiles/zdl.dir/zdl_autogen/mocs_compilation.cpp.o
In file included from /home/kinker31/Doom/qzdl/build/zdl_autogen/mocs_compilation.cpp:15:
/home/kinker31/Doom/qzdl/build/zdl_autogen/EWIEGA46WW/moc_ZDLQSplitter.cpp: In static member function ‘static void ZDLQSplitter::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)’:
/home/kinker31/Doom/qzdl/build/zdl_autogen/EWIEGA46WW/moc_ZDLQSplitter.cpp:79:46: error: cannot convert from pointer to base class ‘QObject’ to pointer to derived class ‘ZDLQSplitter’ via virtual base ‘ZDLWidget’
   79 |     auto *_t = static_cast<ZDLQSplitter *>(_o);
      |                                              ^
make[2]: *** [CMakeFiles/zdl.dir/build.make:85: CMakeFiles/zdl.dir/zdl_autogen/mocs_compilation.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:90: CMakeFiles/zdl.dir/all] Error 2
make: *** [Makefile:91: all] Error 2
```

As an additional note, cmake gives this when generating the makefiles:
```
CMake Warning (dev) at /usr/share/cmake/Modules/FetchContent.cmake:1953 (message):
  Calling FetchContent_Populate(inih) is deprecated, call
  FetchContent_MakeAvailable(inih) instead.  Policy CMP0169 can be set to OLD
  to allow FetchContent_Populate(inih) to be called directly for now, but the
  ability to call it with declared details will be removed completely in a
  future version.
Call Stack (most recent call first):
  CMakeLists.txt:18 (FetchContent_Populate)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@ FetchContent_Declare(
 )
 FetchContent_GetProperties(inih)
 if (NOT inih_POPULATED)
-	FetchContent_Populate(inih)
+	FetchContent_MakeAvailable(inih)
 endif()
 
 add_executable(

--- a/ZDLDMFlagManager.cpp
+++ b/ZDLDMFlagManager.cpp
@@ -21,8 +21,12 @@
 ZDLDMFlagManager::ZDLDMFlagManager(QWidget *parent) :QObject(parent){
 }
 
-void ZDLDMFlagManager::addCheckbox(ZDLDMFlagCheckbox* box){
+void ZDLDMFlagManager::addCheckbox(ZDLDMFlagCheckbox* box) {
+#if QT_VERSION < QT_VERSION_CHECK(6, 7, 0)
 	connect(box, &ZDLDMFlagCheckbox::stateChanged, this, &ZDLDMFlagManager::stateChanged);
+#else
+	connect(box, &ZDLDMFlagCheckbox::checkStateChanged, this, &ZDLDMFlagManager::stateChanged);
+#endif
 	checks.append(box);
 }
 

--- a/ZDLQSplitter.h
+++ b/ZDLQSplitter.h
@@ -16,7 +16,6 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
  
-#include <QMetaObject>
 #ifndef _ZQSPLIT_H_
 #define _ZQSPLIT_H_
 #include <QObject>
@@ -24,7 +23,7 @@
 #include "ZDLWidget.h"
 
 
-class ZDLQSplitter: public virtual ZDLWidget{
+class ZDLQSplitter: public ZDLWidget{
 Q_OBJECT
 	public:
 		ZDLQSplitter(ZDLWidget *parent);

